src/main/java/org/broadinstitute/hellbender/tools/dataflow/MarkDuplicatesDataflow.java

```diff
@@ -344,9 +622,15 @@ private static int score(final GATKRead record) {
     /**
      * Struct-like class to store information about the paired reads for mark duplicates.
      */
-    private static final class PairedEnds {
+    private static final class PairedEnds implements OpticalDuplicateFinder.PhysicalLocation {
```

Comment:
I know. I tried a whole bunch of ways to reengineer it and this seemed to be the path of least code upheaval. This is how the non-dataflow version does it. Otherwise, there is all sorts of copying going on and I was worried about performance. But if you prefer, I can give that a shot, with a note that we could return to this method if it turns out to be better.